File: Spring/src/main/java/ekke/spring/dao/entity/Reservation.java
package ekke.spring.dao.entity;

import ekke.spring.common.BaseEntity;
import lombok.*;

import javax.persistence.*;
import java.time.LocalDate;
import java.util.Date;

@Entity(name = "reservation")
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class Reservation extends BaseEntity {

    @ManyToOne
    @JoinColumn(name = "userId", referencedColumnName = "id")
    private User person;

    @Column(nullable = false, name = "seatNumber")
    private Integer seatNumber;

    @OneToOne
    @JoinColumn(name = "cityId", referencedColumnName = "id")
    private City city;

    @OneToOne
    @JoinColumn(name = "restaurantId",referencedColumnName = "id")
    private Restaurant restaurant;

    @Column(nullable = false, name = "time")
    private LocalDate time;
}

File: Mobile/app/src/main/java/com/restaurant/app/mobile/dto/Reservation.kt
package com.restaurant.app.mobile.dto

import org.json.JSONObject
import java.util.Date
import  java.io.*
import java.time.LocalDate

class Reservation : JSONObject(), Serializable {

    var id: Long = 0

    var userId: Long = 0

    var seatNumber: Int = 0

    var cityId: Long = 0

    var restaurantId: Long = 0

    var time: LocalDate = LocalDate.now()
}